docs/dificil/segment-trees.md

```diff
@@ -6,48 +6,45 @@ tags:
 
 **Autor**: Alexandru Toma
 
-In competitiile de informatica si olimpiada se intalnesc adesea probleme care implica operarea pe subsecventele unui vector. Spre exemplu, cum putem calcula eficient suma, maximul sau minimul pe o subsecventa a unui vector, mai ales cand acesta sufera **modificari in timp real**? 
+În competițiile de informatică și olimpiadă se întâlnesc adesea probleme care implică operarea pe subsecvențele unui vector. Spre exemplu, cum putem calcula eficient suma, maximul sau minimul pe o subsecvență a unui vector, mai ales când acesta suferă **modificări în timp real**? 
 
 !!! note "Observație"
-    Dacă vectorul este static (nu suferă modificări), putem utiliza tehnici simple precum vectori de [sume parțiale](https://roalgo-discord.github.io/arhiva-educationala/usor/partial-sums/?h=sume) sau [RMQ](https://roalgo-discord.github.io/arhiva-educationala/rmq.md) (Range Minimum Query). Totuși, în scenariile unde vectorul se modifică frecvent, aceste tehnici devin ineficiente. Operațiile de query au o complexitate de \(O(1)\) atât pentru sume parțiale, cât și pentru RMQ. Însă, pentru actualizări, trebuie reconstruit tot tabelul de valori, ceea ce duce la o complexitate de \(O(n)\) pentru sume parțiale și \(O(n \log n)\) pentru RMQ.
+    Dacă vectorul este static (nu suferă modificări), putem utiliza tehnici simple precum vectori de [sume parțiale](../ușor/parțial-sums.md) sau [RMQ](../dificil/rmq.md) (Range Minimum Query). Totuși, în scenariile unde vectorul se modifică frecvent, aceste tehnici devin ineficiente. Operațiile de query au o complexitate de \(O(1)\) atât pentru sume parțiale, cât și pentru RMQ. Însă, pentru actualizări, trebuie reconstruit tot tabelul de valori, ceea ce duce la o complexitate de \(O(n)\) pentru sume parțiale și \(O(n \log n)\) pentru RMQ.
 
+**Arborii de intervale** (Segment Trees) reprezintă o soluție elegantă și eficientă pentru acest tip de probleme, permițând efectuarea rapidă de interogări și actualizări în $O(log n)$ pentru ambele operații.
```

Comment:
Probabil ca merge mentionat undeva ca in jargonul romanesc, termenul de aint este utilizat foarte des, cred ca chiar aici ar fi bine.